generator.js

```diff
@@ -1,17 +1,25 @@
+const { normaliseProjectName } = require('./utils');
 module.exports = api => {
   const isTs = api.entryFile.endsWith('.ts')
   const usesRouter = Boolean(require(api.resolve('package.json')).dependencies['vue-router'])
+  const projectName = require(api.resolve('package.json')).name;
 
   api.render({
-    [api.entryFile]: './template/src/main.js'
+    [api.entryFile]: './template/src/main.js',
+    './src/set-public-path.js': './template/src/set-public-path.js',
   }, {
     isTs,
     usesRouter,
-  })
+    appName: projectName,
+  });
 
   api.extendPackage({
     dependencies: {
-      'single-spa-vue': '^1.5.2'
+      'single-spa-vue': '^1.5.2',
+      'systemjs-webpack-interop': '^1.1.0',
+    },
+    scripts: {
+      'build:lib': `vue-cli-service build --target library --name ${normaliseProjectName(projectName)}`,
```

Comment:
I'm not convinced we need this script. The modifications that vue-cli-plugin-single-spa already makes to the webpack config in [index.js](https://github.com/CanopyTax/vue-cli-plugin-single-spa/blob/master/index.js) already set the libraryTarget to UMD which is pretty similar to `--target library`. Also, the `--name` is not needed if you're using [the recommended setup](https://single-spa.js.org/docs/faq/#is-there-a-recommended-setup). SystemJS does not need the UMD bundle to have a name because it already knows the name of the module as specified in the [import map](http://single-spa-playground.org/playground/import-map).